File: Easy/registry/meta_registrydocentiview/meta_registrydocentiview.cs

/*
Easy
Copyright (C) 2021 Università degli Studi di Catania (www.unict.it)
This program is free software: you can redistribute it and/or modify
it under the terms of the GNU General Public License as published by
the Free Software Foundation, either version 3 of the License, or
(at your option) any later version.
This program is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
GNU General Public License for more details.
You should have received a copy of the GNU General Public License
along with this program.  If not, see <http://www.gnu.org/licenses/>.
*/


using System;
using System.Data;
using metadatalibrary;
using metaeasylibrary;
//$CustomUsing$


namespace meta_registrydocentiview
{
    public class Meta_registrydocentiview : Meta_easydata 
	{
        public Meta_registrydocentiview(DataAccess Conn, MetaDataDispatcher Dispatcher) :
            base(Conn, Dispatcher, "registrydocentiview") {
				Name = "Docenti";
			EditTypes.Add("docenti");
			ListingTypes.Add("docenti");
			//$EditTypes$
        }

		private string[] mykey = new string[] {"idreg"};

		public override string[] primaryKey() {
			return mykey;
		}

		//$SetDefault$

		//$Get_New_Row$

		//$IsValid$

		public override void DescribeColumns(DataTable T, string ListingType) {
			base.DescribeColumns(T, ListingType);

			foreach (DataColumn C in T.Columns) {
				DescribeAColumn(T, C.ColumnName, "", -1);
			}
			int nPos = 1;
					
			switch (ListingType) {
				case "docenti": {
						DescribeAColumn(T, "idreg", "Identificativo", nPos++);
						DescribeAColumn(T, "title", "Denominazione", nPos++);
						DescribeAColumn(T, "registry_cf", "Codice fiscale", nPos++);
						DescribeAColumn(T, "registry_p_iva", "Partita iva", nPos++);
						DescribeAColumn(T, "registry_active", "attivo", nPos++);
						DescribeAColumn(T, "registry_docenti_matricola", "Matricola", nPos++);
						DescribeAColumn(T, "sasd_codice", "Codice SASD", nPos++);
						DescribeAColumn(T, "sasd_title", "Settore Artistico Scientifico Disciplinare SASD", nPos++);
						DescribeAColumn(T, "struttura_title", "Denominazione Struttura di afferenza", nPos++);
						DescribeAColumn(T, "struttura_idstrutturakind", "Tipologia Struttura di afferenza", nPos++);
						DescribeAColumn(T, "registryistituti_title", "Istituto, Ente o Azienda", nPos++);
						DescribeAColumn(T, "classconsorsuale_title", "Classe consorsuale", nPos++);
						DescribeAColumn(T, "contrattokind_title", "Tipo", nPos++);
						break;
					}
					//$DescribeAColumn$
			}
		}

		public override string GetSorting(string ListingType) {
			switch (ListingType) {
				case "docenti": {
						return "title asc ";
					}
				//$GetSorting$
			}
			return base.GetSorting(ListingType);
		}

		//$GetStaticFilter$
		
		//$CustomCode$
    }
}

File: Easy/Portale/client/app_segreterie/metadata/registry/meta_registrydocentiview.js
(function() {

    var MetaData = window.appMeta.MetaSegreterieData;

    function meta_registrydocentiview() {
        MetaData.apply(this, ["registrydocentiview"]);
        this.name = 'meta_registrydocentiview';
    }

    meta_registrydocentiview.prototype = _.extend(
        new MetaData(),
        {
            constructor: meta_registrydocentiview,
			superClass: MetaData.prototype,

			describeColumns: function (table, listType) {
				var nPos=1;
				var self = this;
				_.forEach(table.columns, function (c) {
					self.describeAColumn(table, c.name, '', null, -1, null);
				});
				switch (listType) {
					default:
						return this.superClass.describeColumns(table, listType);
					case 'docenti':
						this.describeAColumn(table, 'idreg', 'Identificativo', null, 1000, null);
						this.describeAColumn(table, 'sasd_codice', 'Codice SASD', null, 2200, 50);
						this.describeAColumn(table, 'sasd_title', 'Denominazione SASD', null, 2300, 255);
						this.describeAColumn(table, 'title', 'Denominazione', null, 3000, 101);
						this.describeAColumn(table, 'struttura_title', 'Denominazione Struttura di afferenza', null, 3100, 1024);
						this.describeAColumn(table, 'strutturakind_title', 'Tipologia Tipo Struttura di afferenza', null, 3220, 50);
						this.describeAColumn(table, 'registry_cf', 'Codice fiscale', null, 4000, 16);
						this.describeAColumn(table, 'registryistituti_title', 'Istituto, Ente o Azienda', null, 4100, 101);
						this.describeAColumn(table, 'registry_p_iva', 'Partita iva', null, 5000, 15);
						this.describeAColumn(table, 'classconsorsuale_title', 'Classe consorsuale', null, 5100, 50);
						this.describeAColumn(table, 'registry_active', 'attivo', null, 6000, null);
						this.describeAColumn(table, 'XXregistrylegalstatus', 'Servizi di ruolo - Contratti', null, 51000, null);
//$objCalcFieldConfig_docenti$
						break;
//$objCalcFieldConfig$
				}
				return appMeta.Deferred("describeColumns").resolve();
			},


			//$setCaptions$

			primaryKey: function () {
				return ["idreg"];
			},


			//$isValidFunction$

			//$getStaticFilter$

			getSorting: function (listType) {
				switch (listType) {
					case "docenti": {
						return "title asc ";
					}
					//$getSortingin$
				}
				return this.superClass.getSorting(listType);
			}

        });

    window.appMeta.addMeta('registrydocentiview', new meta_registrydocentiview('registrydocentiview'));

	}());
